### PR TITLE
Odo update refactor

### PR DESCRIPTION
**Pull request by spellingcat**
this redoes the updateOdometry method to actually use the samples we're getting from the other thread, rather than what's passed through the inputs, hopefully makes setPose() actually work, and generally make things clearer.

seems to work in sim but will need to test on the bot once the odo thread is ready

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java [hunk 1]
```diff
@@ -540,7 +539,8 @@ public ChassisSpeeds getRobotRelativeSpeeds() {
   /** Returns the current odometry pose. */
   @AutoLogOutput(key = "Odometry/Robot")
   public Pose2d getPose() {
-    return estimator.getEstimatedPosition();
+    // return estimator.getEstimatedPosition(); //Vision
+    return pose;
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java:543]: do we want this change?

**Review by Lewis-Seiden — CHANGES_REQUESTED** [review]
make sure to merge latest main in, maybe even post-sfr-cleanup since thats going into main soon

File: src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java (hunk 1 above)
Comment (spellingcat) [reply src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java:543]: you would probably know that better than me

**Comment by spellingcat** [comment]
looks like this worked on the robot today

**Pull request merged**